### PR TITLE
pass client and user data to error handler

### PR DESCRIPTION
**Pull request by plicease**
Would it be possible to pass in the client object and some user data into the error handler?  This will help me route errors to the appropriate client object.

### CODE REVIEW

**Comment by kaniini** [comment]
It looks good but can you update the doc comments as well?

File: libpkgconf/libpkgconf.h [hunk 1]
```diff
@@ -144,6 +144,7 @@ struct pkgconf_client_ {
 
 	pkgconf_list_t global_vars;
 
+        void *error_handler_data;
```
Comment (kaniini) [inline libpkgconf/libpkgconf.h]: indentation should be tab, not spaces.

File: libpkgconf/client.c [hunk 2]
```diff
@@ -41,8 +41,9 @@
  *    :return: nothing
```
Comment (kaniini) [inline libpkgconf/client.c]: Please add the new `:param:` here.

File: libpkgconf/client.c [hunk 3]
```diff
@@ -62,10 +63,10 @@ pkgconf_client_init(pkgconf_client_t *client, pkgconf_error_handler_func_t error
  *    :rtype: pkgconf_client_t*
```
Comment (kaniini) [inline libpkgconf/client.c]: Likewise.  Please add the new `:param:` here.

**Comment by plicease** [comment]
fixed!

**Comment by kaniini** [comment]
Coooool.  Thanks.

**Pull request merged**